internal/controlplane/handlers_datasource.go

```diff
@@ -36,7 +36,7 @@ func (s *Server) CreateDataSource(ctx context.Context,
 	}
 
 	// Process the request
-	ret, err := s.dataSourcesService.Create(ctx, dsReq, nil)
+	ret, err := s.dataSourcesService.Create(ctx, uuid.Nil, dsReq, nil)
```

Comment:
Should we leave a TODO about verifying the subscription before doing an update?

Reply:
I think we already do that in the service, I've tagged you in a comment below to verify